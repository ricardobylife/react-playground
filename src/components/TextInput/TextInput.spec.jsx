import { render, screen } from "@testing-library/react"
import userEvent from "@testing-library/user-event";
import { TextInput } from '.'

describe('<TextInput />', () => {
  it('should have a value of searchValue', () => {
    const fn = jest.fn();
    render(<TextInput handleChange={fn} searchValue={'testando'} />);
    const input = screen.getByPlaceholderText('Type your search here...');
    expect(input).toBeInTheDocument();
    expect(input.value).toBe('testando');
  })

  it('should call handleChange function on each key press', () => {
    const fn = jest.fn();
    render(<TextInput handleChange={fn} />);
    const input = screen.getByPlaceholderText('Type your search here...');

    const value = 'o valor da busca';
    
    userEvent.type(input, value)
    
    expect(input.value).toBe(value);
    expect(fn).toHaveBeenCalledTimes(value.length);

    input.value = '';

    userEvent.type(input, '')
    expect(input.value).toBe('');
    expect(fn).toHaveBeenCalledTimes(value.length);

  })

  it('should macth the snapshot', () => {
    const fn = jest.fn();
    const {container} = render(<TextInput type="search" placeholder="Type your search here" onChange={fn} />);
    
    expect(container.firstChild).toMatchSnapshot();
  })

});
