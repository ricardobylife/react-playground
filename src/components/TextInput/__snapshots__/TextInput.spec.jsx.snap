// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput /> should macth the snapshot 1`] = `
<input
  class="text-input"
  placeholder="Type your search here..."
  type="search"
  value=""
/>
`;
